### PR TITLE
Adding on_error and on_abort notifications to slack.

### PR DESCRIPTION
**Pull request by Ardiea**
#### Pre-Flight checklist

- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
#1498 

#### What's this PR do?
This will notify our slack channel when a course job ends in an 'error' condition or if it is aborted by a concourse user. 

#### How should this be manually tested?
N/A

#### Where should the reviewer start?
(Optional)

#### Any background context you want to provide?
(Optional)

#### Screenshots (if appropriate)
(Optional)

#### What GIF best describes this PR or how it makes you feel?
(Optional)

### CODE REVIEW

File: content_sync/pipelines/definitions/concourse/site-pipeline.yml [hunk 1]
```diff
@@ -107,6 +107,22 @@ jobs:
               params:
                 alert_type: failed
                 text: "Failed on get webpack-json : ((pipeline_name))/((site-name))"
+        on_error:
```
Comment (Ardiea) [inline content_sync/pipelines/definitions/concourse/site-pipeline.yml:110]: @mbertrand So as I was finishing this up, I got to thinking, should we add calls to 'ocw-studio-webhook' to the `on_error` and `on_abort` clauses as well? I can't think of a reason _not_ to but I'm not familiar with the app side of equation.

Comment (mbertrand) [reply content_sync/pipelines/definitions/concourse/site-pipeline.yml:110]: Sure, that would be great!  The `status` sent in the webhook can be `errored` for `on_error` and `aborted` for `on_abort`.

Comment (Ardiea) [reply content_sync/pipelines/definitions/concourse/site-pipeline.yml:110]: Okay I will get started!

Comment (Ardiea) [reply content_sync/pipelines/definitions/concourse/site-pipeline.yml:110]: Should I switch the `status` in the `on_failure` webhook calls to `failed` or should it remain as `errored`?

Comment (mbertrand) [reply content_sync/pipelines/definitions/concourse/site-pipeline.yml:110]: Leave it as `errored`.  There is no constant for `failed` in the app currently, and I'm not sure if making a distinction between `errored` and `failed` would be useful.

https://github.com/mitodl/ocw-studio/blob/c8da66e64fe2e2c4b449bdf99f66d5c4a9459da8/websites/constants.py#L92-L110

**Review by mbertrand — APPROVED** [review]
👍

**Pull request merged**